lib/cache/resource.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+
+var _ = require('lodash');
+var MemoryCache = require('./memory');
+
+function ResourceCache(cacheId, options) {
+    this._cache = new MemoryCache(cacheId, options);
+}
+
+ResourceCache.prototype.get = function(cacheKey) {
+    return getFromCache.call(this, cacheKey);
+};
+
+function getFromCache(cacheKey) {
+    var entry = _.cloneDeep(this._cache.get(cacheKey));
+
+    // If parent or nested resources not in cache then return undefined.
```

Comment:
Can you explain the reasoning behind this? I'm not sure if this is necessary. For example, here's a sample response from Monocle that contains a top-level object and a nested resource:

    {
        displayName: "Hector V",
        age: 33,
        primaryPhoto: {
            id: 71334444,
            height: 720,
            urlTemplate: "http://i0.stage.tagstat.com/p1/%s/f83vBfqJJGvJuYA6PjYtoDdVk8tUUpNlgBX09YpCtytAUVwVe3K4Ag==.jpg",
            width: 720,
            $id: "/users/123598901406/photo/71334444",
            $expires: 300000
        },
        gender: "M",
        $id: "/users/123598901406",
        $expires: 60000
    }

In this example, would the photo expire when the user data expires?